### PR TITLE
Additional external events

### PR DESCRIPTION
**Pull request by seanfarmar**
Related to #32

### CODE REVIEW

File: src/ServiceControl.Contracts/FailedMessagesUnArchived.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    public class FailedMessagesUnArchived
+    {
+        /// <summary>
+        /// The number of messages that failed and were un-archived
+        /// </summary>
+        public int MessagesCount { get; set; }
+        /// <summary>
+        /// A list of the unique identifiers of the messages that failed and were un-archived
+        /// </summary>
+        public string[] DocumentIds { get; set; }
```
Comment (danielmarbach) [inline src/ServiceControl.Contracts/FailedMessagesUnArchived.cs:15]: Shouldn't this be called either `FailedMessageIds` or `MessageIds`?

**Review by tmasternak — APPROVED** [review]

File: src/ServiceControl.Contracts/FailedMessageArchived.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿namespace ServiceControl.Contracts
+{
+    /// <summary>
+    /// Event emitted for each failed message that was archived
+    /// </summary>
+    public class FailedMessageArchived
```
Comment (mikeminutillo) [inline src/ServiceControl.Contracts/FailedMessageArchived.cs:6]: We usually do bulk archiving of groups. Would it make more sense to emit a single event with a list of failed message ids?

Comment (seanfarmar) [reply src/ServiceControl.Contracts/FailedMessageArchived.cs:6]: Would this be emitted with the group archived event?

**Comment by seanfarmar** [comment]
opened a new PR taken off the pre 2.0 tag #45

**Pull request closed without merging**